backend/sirius-components-diagrams-tests/src/main/java/org/eclipse/sirius/components/diagrams/tests/NodeAssert.java

```diff
@@ -121,4 +121,47 @@ public NodeAssert hasBounds(double x, double y, double width, double height) {
 
         return this;
     }
+
+    public NodeAssert hasEveryChildWithinItsBounds() {
+        this.isNotNull();
+        assertThat(this.actual.getChildNodes()).allMatch(child -> this.isChildWithinParentBounds(this.actual, child));
+        return this;
+    }
```

Comment:
Unless I'm missing something: 
- this algorithm is not recursive so a grand child could be outside of its grand parent without issues
- The border nodes of the child nodes are not considered, they could be outside of the parent node